app/services/subscription_validator.rb

```diff
@@ -76,10 +76,11 @@ def schedule_allowed?
   end
 
   def credit_card_ok?
-    return unless payment_method.andand.type == "Spree::Gateway::StripeConnect"
-    return errors.add(:credit_card, :blank) unless credit_card_id
-    return if customer.andand.user.andand.credit_card_ids.andand.include? credit_card_id
-    errors.add(:credit_card, :not_available)
+    return unless customer && payment_method
+    return unless payment_method.type == "Spree::Gateway::StripeConnect"
+    return errors.add(:credit_card, :charges_not_allowed) unless customer.allow_charges
+    return if customer.user.andand.default_card.present?
```

Comment:
We should add the constraint on customers to have a user always. Having this conditional everytime we use a customer object adds unnecessary complexity.

Reply:
I tend to agree, but I think that is outside the scope of this work.

Reply:
@sauloperez I watched the Rails videos you shared on Slack and one interesting idea there was to return a `Guest` user class when there is no registered user. It would respond to the same methods. In this case, `Guest#default_card` would always return `nil`.